plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TypeConverter.java

```diff
@@ -153,14 +168,27 @@ public static org.apache.iceberg.types.Type toIcebergType(Type type)
         if (type.equals(UUID)) {
             return Types.UUIDType.get();
         }
-        if (type instanceof RowType) {
-            return fromRow((RowType) type);
-        }
-        if (type instanceof ArrayType) {
-            return fromArray((ArrayType) type);
+        if (columnIdentity.isPresent()) {
+            if (type instanceof RowType) {
+                return fromRow((RowType) type, columnIdentity.get());
```

Comment:
can we use `fromX(type, Optional<ColumnIdentity>)` syntaxes here where X is Row/Map/Array ? That'd remove the duplication within if/else - and also eliminate method overloading for `fromX(type, ColumnIdentiy), fromX(type)` to only `fromX(type, Optional<ColumnIdentity>)`?

Reply:
I implemented it like you said, but it is disassembled here to distinguish the two logics, but as @alexjo2144 said in  https://github.com/trinodb/trino/pull/14837#discussion_r1015634415 , I I separated the two parts, maybe I get wrong, I will roll back it